pkg/agent/openflow/client.go

```diff
@@ -88,11 +88,18 @@ type Client interface {
 
 	// InstallServiceGroup installs a group for Service LB. Each endpoint
 	// is a bucket of the group. For now, each bucket has the same weight.
-	InstallServiceGroup(groupID binding.GroupIDType, withSessionAffinity bool, endpoints []proxy.Endpoint) error
+	InstallServiceGroup(groupID binding.GroupIDType, endpoints []proxy.Endpoint) error
 	// UninstallServiceGroup removes the group and its buckets that are
 	// installed by InstallServiceGroup.
 	UninstallServiceGroup(groupID binding.GroupIDType) error
 
+	// InstallEndpointRegSetFlows installs flows that set registers to make service group action shorter, thus more items
+	// can be included in a add-group message.
+	InstallEndpointRegSetFlows(withSessionAffinity bool, clusterIP net.IP, port int, endpoints []proxy.Endpoint) error
```

Comment:
Do we need another table and flows for it?
I was thinking just moving the common actions to table 41 like below:
```
table=41, priority=200,tcp,reg4=0x10000/0x70000,nw_dst=10.96.0.1,tp_dst=443 actions=group:8,load:0x2->NXM_NX_REG4[16..18],load:0x1->NXM_NX_REG0[19],resubmit(,42)

group_id=8,type=select,bucket=bucket_id:0,weight:100,actions=load:0xac10000b->NXM_NX_REG3[],load:0x192b->NXM_NX_REG4[0..15]
```

Could it work?

Reply:
Unfortunately, this doesn't work as we expected. I also verify that from OVS documents. https://man7.org/linux/man-pages/man7/ovs-actions.7.html

```
12. group, output, resubmit, ct_clear, or ct. If more than
                  one of these actions is present, then the one listed
                  earliest above is executed and the others are ignored,
                  regardless of the order in which they were added to
                  the action set. (If none of these actions is present,
                  the action set has no real effect, because the
                  modified packet is not sent anywhere and thus the
                  modifications are not visible.)
```